### PR TITLE
Update VS Code API to version 1.46.0

### PR DESCRIPTION
**Pull request by arekzaluski**
Update VSCode version to version 1.46.0. It is needed for `vscode-clangd` plugin to work. We do not need to upstream it to Theia as the current master already has a newer version (1.50.0). I've updated to a minimal version that is needed for vscode-clangd to work.

### CODE REVIEW

File: packages/plugin-ext/src/plugin/plugin-context.ts [hunk 1]
```diff
@@ -789,7 +789,7 @@ export function createAPIFactory(
         };
 
         return <typeof theia>{
-            version: require('../../package.json').version,
+            version: process.env['VSCODE_API_VERSION'] || require('../../package.json').version,
```
Comment (mcgordonite) [inline packages/plugin-ext/src/plugin/plugin-context.ts:792]: Why is this needed? I think that's the version of the Theia plugin API rather than the VSCode version.

Comment (arekzaluski) [reply packages/plugin-ext/src/plugin/plugin-context.ts:792]: Unfortunately, it is.

Comment (arekzaluski) [reply packages/plugin-ext/src/plugin/plugin-context.ts:792]: You can even see this change in master of theia. Here: https://github.com/eclipse-theia/theia/blob/e33708b77117a044f5f0a3fa97309c8b9fe3664f/packages/plugin-ext-vscode/src/node/plugin-vscode-init.ts#L56

Comment (arekzaluski) [reply packages/plugin-ext/src/plugin/plugin-context.ts:792]: To be honest I'm not 100% sure why Theia doesn't need it

Comment (arekzaluski) [reply packages/plugin-ext/src/plugin/plugin-context.ts:792]: That being said they haven't updated to a new version of vscode-clangd yet

Comment (arekzaluski) [reply packages/plugin-ext/src/plugin/plugin-context.ts:792]: Without it an incorrect VSCode version is detected as this code is used to create a default version value. It means that vscode-clangd is not loaded.

Comment (arekzaluski) [reply packages/plugin-ext/src/plugin/plugin-context.ts:792]: I get a feeling that for some reason plugins are incorrectly marked and they are loaded as theia plugins instead of vscode plugins.

Comment (mcgordonite) [reply packages/plugin-ext/src/plugin/plugin-context.ts:792]: Interesting, it looks like the plugin is correctly recognized as a VSCode extension by the plugin-ext-vscode package, but the version check in the vscode language server code seems to get the wrong `vscode.version`.

**Review by mcgordonite — APPROVED** [review]
LGTM. The plugin-context version change looks like it's required for now, but it would help to open an issue on upstream Theia.

**Comment by thegecko** [comment]
@arekzaluski does this require merging or deleting?
